index.html

```diff
@@ -687,9 +687,28 @@ <h4 class="modal-title" i18n="modal.layer.edit">Edition de couche</h4>
                                             </div>
                                         </div>
                                         <label for="frm-sld" i18n="modal.layer.sld.external">SLD externe</label>
-                                        <input type="text" class="form-control checkedurl" id="frm-sld"
+                                        <div class="input-group">
+                                            <input type="text" class="form-control checkedurl" id="frm-sld"
                                                placeholder="URL SLD externe" i18n="modal.layer.display.ph.external_sld"/>
+                                            <span class="input-group-btn">
+                                                <input type="file" id="frm-upload-sld-file" style="display: none;" onchange="uploadSldFileToBackend(this)">
+                                                <button class="btn btn-default"
+                                                        type="button"
+                                                        i18n="modal.layer.uploadsldfile"
+                                                        onclick="document.getElementById('frm-upload-sld-file').click()"
+                                                >
+                                                    Téléverser un SLD
+                                                </button>
+                                                <button class="btn btn-default" type="button">
+                                                    <span class="glyphicon glyphicon-info-sign"
+                                                          data-toggle="tooltip"
+                                                          data-placement="top"
+                                                          title="Vous pouvez insérer une URL vers un SLD. Si votre couche provient d'un geoserver, vous pouvez specifier uniquement le nom du SLD dans geoserver (ex: un_entrepot:mon_sld)"
```

Comment:
same comment for i18n ?

Reply:
I have no idea how to make this work with i18njs though

Reply:
10 lines above ;)

https://github.com/geobretagne/mviewerstudio/blob/aaaae8e45ad927f8e509c87fda319c3fb32f654a/index.html#L697